mypy/plugins/dataclasses.py

```diff
@@ -341,6 +345,35 @@ def transform(self) -> bool:
 
         return True
 
+    def _add_internal_replace_method(self, attributes: list[DataclassAttribute]) -> None:
+        """
+        Stashes the signature of 'dataclasses.replace(...)' for this specific dataclass
+        to be used later whenever 'dataclasses.replace' is called for this dataclass.
+        """
+        arg_types: list[Type] = [Instance(self._cls.info, [])]
```

Comment:
Btw this (and also the return type below, etc) look not very careful about generic dataclasses.

Reply:
You're right. I'd be better off not adding the first arg and ret_type in this phase, and instead add them in the hook:
c0058950753ef275edb6abd6775660a4b0dc971e